### PR TITLE
feat(service/rcp): unified error handling

### PR DESCRIPTION
**Pull request by tzdybal**
* error handling is unified for all error types and endpoints
* duplicated code is extracted into function
* errors are returned as valid JSON, not plain text

### CODE REVIEW

**Review by liamsi — COMMENTED** [review]
lgtm but I think @renaynay should have the final say here.

File: service/rpc/util.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package rpc
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func writeError(w http.ResponseWriter, statusCode int, endpoint string, err error) {
+	w.WriteHeader(statusCode)
+	errBody, _ := json.Marshal(err.Error())
```
Comment (renaynay) [inline service/rpc/util.go:10]: Why not check for err here?

File: service/rpc/util.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	_, werr := w.Write(errBody)
+	if werr != nil {
+		log.Errorw("writing response", "endpoint", namespacedSharesEndpoint, "err", werr)
```
Comment (renaynay) [inline service/rpc/util.go:13]:
```suggestion
		log.Errorw("writing response", "endpoint", endpoint, "err", werr)
```
```suggestion
		log.Errorw("writing response", "endpoint", namespacedSharesEndpoint, "err", werr)
```

Comment (renaynay) [inline service/rpc/util.go:13]:
```suggestion
		log.Errorw("writing response", "endpoint", endpoint, "err", werr)
```

File: service/rpc/util.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	}
+	log.Errorw("serving request", "endpoint", namespacedSharesEndpoint, "err", err)
```
Comment (renaynay) [inline service/rpc/util.go:15]:
```suggestion
	log.Errorw("serving request", "endpoint", endpoint, "err", err)
```

Comment (renaynay) [inline service/rpc/util.go:15]:
```suggestion
	log.Errorw("serving request", "endpoint", endpoint, "err", err)
```
```suggestion
	log.Errorw("serving request", "endpoint", namespacedSharesEndpoint, "err", err)
```

File: service/rpc/util.go (hunk 1 above)
Comment (tzdybal) [reply service/rpc/util.go:10]: I assumed that we can be pretty sure about marshaling a string. I can add this.

Comment (renaynay) [reply service/rpc/util.go:10]: Yeah let's add the err check too just in case. Otherwise good to go

**Comment by tzdybal** [comment]
Added error handling and rebased the code.

**Comment by renaynay** [comment]
Can I close @tzdybal ?

**Comment by tzdybal** [comment]
Yeah, ofc.

**Pull request closed without merging**